### ISSUE TITLE
Script to run the tests locally such that they all pass

### ISSUE THREAD

**Issue by dblalock** [mosaicml/composer#367]
## Motivation

If a user downloads composer and wants to either:

1. make sure everything installed okay, or
2. try modifying it without breaking everything (e.g., to add a new method), 

it's really helpful to be able to run the full test suite and know whether failures are because of your code. I know this is helpful (at least for case 2) based on my experience developing methods.

Also, having a command to run the tests post-install is just a pretty common element of installation documentation.

## [Optional] Implementation

Below is equivalent to what I run to get everything passing locally. I say "equivalent" because I don't setup the environment and reinstall pillow-simd regularly; I just run the last command.

The sketchy part here is getting pillow-simd installed so that test_pillow_simd passes. Might be more robust to just have it skip that test.
```
# this block is so that test_pillow_simd passes
pip uninstall pillow
pip uninstall pillow-simd 
pip install pillow-simd  # needs certain deps; might choke

# this block is to avoid a few torch.distributed failures in test_checkpoint, test_ddp, and a couple other files
export NODE_RANK=0
export LOCAL_RANK=0
export LOCAL_WORLD_SIZE=1

# actual testing command
pytest -s --tb=short --disable-pytest-warnings --test_duration all tests/
```

## Additional context

`bash scripts/test.sh` doesn't currently allow the desired behavior :

```
~/code/composer $ bash scripts/test.sh                                              
+ JUNIT_PREFIX=build/output/composer
++ dirname build/output/composer
+ mkdir -p build/output
+ python -m composer.cli.launcher -n 1 --master_port 26000 -m coverage run -m pytest --junitxml build/output/composer.n1.junit.xml
/Users/davis/.pyenv/versions/3.8.6/bin/python: No module named coverage
Waiting up to 30 seconds for all training processes to terminate...
Process 1566 excited with code 1
ERROR:__main__:Subprocess 1566 exited with code 1
```

**Comment by ravi-mosaicml** [comment]
Great points raised here! Can you add this to the `CONTRIBUTING.md` document?

If you are developing on composer, one should install `pip install mosaicml[dev]`

> # this block is to avoid a few torch.distributed failures in test_checkpoint, test_ddp, and a couple other files
> export NODE_RANK=0
> export LOCAL_RANK=0
> export LOCAL_WORLD_SIZE=1

@jbloxham can you create a PR to fix these tests? I think we'll need to guard torch.distributed initialization unless if all these variables are set.

**Comment by ravi-mosaicml** [comment]
#389 Fixes this

> ```shell
> # this block is to avoid a few torch.distributed failures in test_checkpoint, test_ddp, and a couple other files
> ```

**Comment by dblalock** [comment]
Moved @ravi-mosaicml's request for CONTRIBUTING.md notes to #440. I'm calling this issue addressed by #387.
